### ISSUE TITLE
willTransitionTo is not triggered when using Router API

### ISSUE THREAD

**Issue by vasya10** [rackt/react-router#931]
Using react-router 0.12.4.

I have a simple ReactJS component:

```javascript
var Dashboard = React.createClass({

  statics: {
    willTransitionTo: function (transition) {
      console.log("intercepting transition path", transition.path);
      console.log("verifying user logged in");
    }
  },
  getInitialState: function () {
    return {items:[]};
  },
  render: function () {
    return (
      <div>Dashboard Page</div>
    );
  }
});
```
My routes are defined programmatically like this:
```javascript
var MainRoute = Router.createRoute( { name: “app", path: “/", handler: Main });
var DashboardRoute = Router.createRoute( { name: “dashboard", path: “/dashboard", parentRoute: MainRoute, handler: Dashboard });
Router.run([MainRoute], function (Handler) {
  React.render(<Handler/>, document.body);
});
```
When I do like above, willTransitionTo is not fired (I dont see the console log). 
But if I change the above code to use the jsx technique, like below, it works:
```javascript
var routes = (
  <Route name="app" path="/" handler={Main}>
    <Route name=“Dashboard" path=“/dashboard" handler={Dashboard} />
  </Route>
);
Router.run(routes, function (Handler) {
  React.render(<Handler/>, document.body);
});
```
Am I missing any parameter for API access that triggers willTransitionTo?

**Comment by taurose** [comment]
I don't know about the reasoning behind this, but it seems you need to use `opts.onEnter` and `opts.onLeave`. See https://github.com/rackt/react-router/blob/v0.12.4/modules/createRoutesFromReactChildren.js#L29

**Comment by vasya10** [comment]
@taurose Thanks, yes that worked, I set it to this:

```javascript
Router.createRoute({"Dashboard" path="/dashboard" handler=Dashboard  onEnter=Dashboard.willTransitionTo});
```

I looked at the code a bit. The JSX technique uses the ReactElement and goes via createRouter which calls the createRoutesFromChildren, which automatically sets the the willTransitionTo, if handler is present (https://github.com/rackt/react-router/blob/v0.12.4/modules/createRoutesFromReactChildren.js#L28) 
[side note: i've not tested this, but just from code: there may be a problem here, if I explicitly set a handler for onEnter in jsx, it may still get overwritten with willTransitionTo, because there is no check for value set.]

The API directly uses the createRoute(options) method, where the new routes is created at https://github.com/rackt/react-router/blob/v0.12.4/modules/Route.js#L88. This does not default-set willTransitionTo.

Just for consistency between Route API and the Route JSX, it may be useful to add the same default code there.
